pkg/vsphere/tasks/waiter.go

```diff
@@ -89,6 +102,56 @@ func WaitForResult(ctx context.Context, f func(context.Context) (Task, error)) (
 	}
 }
 
+func fixTask(ctx context.Context, sess *session.Session, target object.Reference) error {
+	switch vmm := target.(type) {
+	case *object.VirtualMachine:
+		vm := vm.NewVirtualMachine(ctx, sess, target.Reference())
```

Comment:
I think its better if we do it now, since changes to WaitForResult result in a lot of churn.